src/script/util/TypedEventTarget.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+export class TypedEventTarget<EventDef extends {type: any}> extends EventTarget {
+  public dispatch(event: EventDef): boolean {
+    return this.dispatchEvent(new Event(event.type));
```

Comment:
this in this case refers to eventTarget? this is one of those times I find `this` to be a little ambiguous

Reply:
Yeah, it refers to the native class. Would you do something different? 
Notice that the children class method is `dispatch` VS the native method is `dispatchEvent`, maybe that's the source of the confusion?

Reply:
i think that is the source of the confusion for me, I don't know if I have any other solution though, I think its fine.